### PR TITLE
python312Packages.reflex: init at 0.5.9

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

https://reflex.dev

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/sqlalchemy-file/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "jowilf";
+    repo = "sqlalchemy-file";
+    rev = version;
+    hash = "sha256-gtW7YA/rQ48tnqPdypMnSqqtwb90nhAkiQNhgEr1M3I=";
+  };
+
+  nativeBuildInputs = [ hatchling ];
```
Comment (natsukium) [inline pkgs/development/python-modules/sqlalchemy-file/default.nix:26]:
```suggestion
  build-system = [ hatchling ];
```

File: pkgs/development/python-modules/sqlalchemy-file/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/sqlalchemy-file/default.nix:28]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/reflex/default.nix [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  disabledTestPaths = [
+    "benchmarks/"
+    "integration/"
+  ];
+
+  pythonImportsCheck = [ "reflex" ];
+
+  meta = with lib; {
+    description = "Web apps in pure Python";
+    homepage = "https://github.com/reflex-dev/reflex";
```
Comment (natsukium) [inline pkgs/development/python-modules/reflex/default.nix:133]:
```suggestion
    homepage = "https://github.com/reflex-dev/reflex";
    changelog = "https://github.com/reflex-dev/reflex/releases/tag/${src.rev}";
```

File: pkgs/development/python-modules/reflex/default.nix [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    license = licenses.asl20;
+    maintainers = with maintainers; [ pbsds ];
```
Comment (natsukium) [inline pkgs/development/python-modules/reflex/default.nix:135]: How about adding `meta.mainProgram`?
https://github.com/reflex-dev/reflex/tree/v0.5.9?tab=readme-ov-file#-create-your-first-app

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 333337` at 6ff9b46c run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.reflex</li>
    <li><a href="https://gist.github.com/natsukium/18d23d41e6b9f596360ac193cfac4aab">python311Packages.starlette-admin</a></li>
    <li>python312Packages.reflex</li>
    <li><a href="https://gist.github.com/natsukium/5c0b947a48bd45d9dc164a1633d8e46d">python312Packages.starlette-admin</a></li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python311Packages.reflex-hosting-cli</li>
    <li>python311Packages.sqlalchemy-file</li>
    <li>python312Packages.reflex-hosting-cli</li>
    <li>python312Packages.sqlalchemy-file</li>
  </ul>
</details>

**Comment by natsukium** [comment]
The errors seem to be caused by darwin sandbox.

**Comment by pbsds** [comment]
> The errors seem to be caused by darwin sandbox.

this maybe?

```patch
--- a/pkgs/development/python-modules/reflex/default.nix
+++ b/pkgs/development/python-modules/reflex/default.nix
@@ -120,10 +120,15 @@ buildPythonPackage rec {
     "test_potentially_dirty_substates"
     # flaky, KeyError: 'reflex___state____state'
     "test_preprocess"
+    # flaky, depends on test order
+    "test_ensuring_pk"
   ];
   disabledTestPaths = [
     "benchmarks/"
     "integration/"
+  ] ++ lib.optionals stdenv.isDarwin [
+    # sandbox issues
+    "tests/sqla/test_sync_engine.py"
   ];
 
   pythonImportsCheck = [ "reflex" ];
```

btw, which fork do you use to post gists of the logs?

**Comment by natsukium** [comment]
> btw, which fork do you use to post gists of the logs?

I use [a fork](https://github.com/natsukium/nixpkgs-review/tree/r-rmcgibbo-1) based on [this PR](https://github.com/Mic92/nixpkgs-review/pull/187).
I would like to submit a PR, but... I haven't yet.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 333337` at c1dfc595 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python311Packages.reflex</li>
    <li><a href="https://gist.github.com/natsukium/a9323ccb24211260293de2e5a6a44b97">python311Packages.sqlalchemy-file</a></li>
    <li>python311Packages.starlette-admin</li>
  </ul>
</details>
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>python311Packages.reflex-hosting-cli</li>
    <li>python312Packages.reflex</li>
    <li>python312Packages.reflex-hosting-cli</li>
    <li>python312Packages.sqlalchemy-file</li>
    <li>python312Packages.starlette-admin</li>
  </ul>
</details>

**Comment by pbsds** [comment]
not sure what the error might be, but if this doesn't work i might just mark it broken on darwin

**Comment by pbsds** [comment]
man i dislike darwin, the sandbox is so flaky...

**Pull request merged**